app/models/miq_policy.rb

```diff
@@ -11,12 +11,18 @@ class MiqPolicy < ApplicationRecord
                                  PhysicalServer
                                  Vm).freeze
 
-  POLICY_TOWHAT_APPLIES_TO_CLASSES = TOWHAT_APPLIES_TO_CLASSES.inject({}) { |hsh, key| hsh[key] = ui_lookup(:model => key); hsh }.freeze
+  def self.policy_towhat_applies_to_classes(opts = nil)
+    policyToWhatAppliesToClasses = TOWHAT_APPLIES_TO_CLASSES.inject({}) { |hsh, key| hsh[key] = ui_lookup(:model => key); hsh }
+    opts.nil? ? policyToWhatAppliesToClasses : policyToWhatAppliesToClasses[opts]
+  end
```

Comment:
as I'm reading this, let's get rid of all this lookup code - if the method returns a hash the caller can just use it like a Hash.

```suggestion
  def self.policy_towhat_applies_to_classes(opts = nil)
    TOWHAT_APPLIES_TO_CLASSES.inject({}) { |hsh, key| hsh[key] = ui_lookup(:model => key); hsh }
  end
```

Reply:
Even cleaner:

```suggestion
  def self.policy_towhat_applies_to_classes(opts = nil)
     TOWHAT_APPLIES_TO_CLASSES.index_with { |key| = ui_lookup(:model => key) }
  end
```